compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -1603,7 +1603,11 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(using Context) {
       res
 
     override def typedTypeApply(tree: untpd.TypeApply, pt: Type)(using Context): Tree =
-      inlineIfNeeded(constToLiteral(betaReduce(super.typedTypeApply(tree, pt))))
+      val tree1 = inlineIfNeeded(constToLiteral(betaReduce(super.typedTypeApply(tree, pt))))
+      if tree1.symbol.isQuote then
+        ctx.compilationUnit.needsStaging = true
```

Comment:
needsStaging seems like a misleading name if it's set to true whenever there is a quote?

Reply:
It is based on the name of the `Staging` phase that it enables.

Reply:
But couldn't we merge needsStaging and needsQuotePickling into one variable which is checked in both phases?

Reply:
I thought that we had some cases that needed quote pickling without requiring staging. It seems that this is no longer the case and we can simplify it to a single `needsStaging` contdition.

Reply:
Simplified